Strip multiple slashes from DNS name.

kierenevans opened the pull request:
Stops `example.com/subfolder/image` becoming `subfolder/image.docker` in the DNS entry.

Instead, it will become `subfolder_image.docker`.

kierenevans committed:
Strip multiple slashes from DNS name.

sroze commented on spec/Dock/Docker/Dns/DnsDockResolverSpec.php:
```diff
@@ -32,4 +32,10 @@ function it_removes_the_slash_in_image_name()
+    function it_removes_more_than_one_slash_in_image_name()
+    {
+        $this->getDnsByContainerNameAndImage('container', 'someone/folder/image:latest')
+            ->shouldContain('container.folder_image.docker');
```
I do think it should container `container.image.docker` and `image.docker`, nothing related with the folder here :)

kierenevans committed:
Use the last part of the image URL/name for the DNS name.

Fix using dashes in hostnames, which Internet Explorer does not like for cookies.

sroze commented:
Yay 👍
